src/middlewares/auth-middleware.ts

```diff
@@ -48,104 +47,24 @@ async function authentication(
-          throw new NoDBUserFoundError();
+        if (request.url && request.url.includes("/api/")) {
+          throw webErrorFactory(webErrors.unauthorized);
         }
-        span.setDisclosedAttribute("user found on DB", true);
-
-        const refreshedTokens = await oauth2Client
-          .refreshTokens(user.refresh_token)
-          .catch((error) => {
-            span.setDisclosedAttribute("is token refreshed", false);
-            if (error instanceof UnreachableError) {
-              throw webErrorFactory({
-                ...webErrors.unreachable,
-                parentError: error,
-              });
-            }
-
-            throw error;
-          });
-        span.setDisclosedAttribute("is token refreshed", true);
-
-        await decryptVerifyAccessToken(refreshedTokens.access_token).catch(
-          (error) => {
-            span.setDisclosedAttribute("is access_token valid", false);
-            if (
-              error instanceof MissingJWKSURIError ||
-              error instanceof InvalidAudienceError ||
-              error instanceof MissingKeyIDHS256Error ||
-              error instanceof AlgoNotSupportedError ||
-              error instanceof InvalidKeyIDRS256Error ||
-              error instanceof UnhandledTokenType
-            ) {
-              throw webErrorFactory({
-                ...webErrors.badRequest,
-                parentError: error,
-              });
-            }
-
-            if (error instanceof UnreachableError) {
-              throw webErrorFactory({
-                ...webErrors.unreachable,
-                parentError: error,
-              });
-            }
-
-            throw error;
-          },
-        );
-        span.setDisclosedAttribute("is refresh access_token valid", true);
-
-        await setServerSideCookies(
-          response,
-          "user-cookie",
-          {
-            access_token: refreshedTokens.access_token,
-            sub,
-          },
-          {
-            shouldCookieBeSealed: true,
-            cookieSalt: configVariables.cookieSalt,
-            maxAge: 24 * 60 * 60,
-            path: "/",
-            httpOnly: true,
-            secure: true,
-          },
-        );
 
-        span.setDisclosedAttribute("is cookie set", true);
-
-        await getAndPutUser(
-          { sub, refresh_token: refreshedTokens.refresh_token },
-          user,
-        ).catch((error) => {
-          span.setDisclosedAttribute("database reachable", false);
-          throw webErrorFactory({
-            ...webErrors.databaseUnreachable,
-            parentError: error,
-          });
-        });
-
-        span.setDisclosedAttribute("user updated on DB", true);
+        const redirectURL = !request.url ? "/account" : request.url;
 
-        return;
+        response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+        response.setHeader(
+          "location",
+          `/api/auth-connect/refresh-token?next=${redirectURL}`,
```

Comment:
does it always work? Shouldn't we use `encodeURIComponent`?

Reply:
It worked when I tested it during development, but always, I don't know. Anyway, I didn't knew `encodeURIComponent` before looking at your comment, and just in case, it seems safer to add it. 

Is it mean to be used like that ? 
`/api/auth-connect/refresh-token?next=${encodeURIComponent(redirectURL)}`

Reply:
yes